src/sentry/integrations/discord/requests/base.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+import dataclasses
+from collections.abc import Mapping
+
+from rest_framework import status
+from rest_framework.request import Request
+
+from sentry import options
+from sentry.services.hybrid_cloud.integration import RpcIntegration, integration_service
+
+from ..utils import logger, verify_signature
+
+
+@dataclasses.dataclass(frozen=True)
+class DiscordRequestError(Exception):
+    """
+    Something was invalid about the request from Discord.
+    Includes the status the endpoint should return, based on the error.
+    """
+
+    status: int
+
+
+class DiscordRequest:
```

Comment:
A lot of this PR is inspired by Slack's implementation. I liked how everything was split up there.

https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/slack/webhooks/event.py
https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/slack/requests/base.py